BDArmory/Control/BDModuleSurfaceAI.cs

```diff
@@ -839,7 +843,7 @@ public override bool IsValidFixedWeaponTarget(Vessel target)
             => !BroadsideAttack &&
             (((target != null ? target.Splashed : false) && (SurfaceType & AIUtils.VehicleMovementType.Water) != 0) //boat targeting boat
```

Comment:
`target != null ? target.Splashed : false` is a weird way of writing `target != null && target.Splashed`. I'd suggest changing this and the next line for the latter, like for the hovercraft check.

Reply:
Legacy Code strikes again